### ISSUE TITLE
Using data getter on empty DataGrid produces key error

### ISSUE THREAD

**Issue by ollyhensby** [bloomberg/ipydatagrid#322]
**Describe the bug**
When there's no data in the grid, the data getter produces an error.

**To Reproduce**
1. Import DataGrid and pandas

```python
from ipydatagrid import DataGrid
import pandas as pd
```

2. Create simple dataframe and input into DataGrid

```python
df = pd.DataFrame([1], columns=["a"])
grid = DataGrid(df)
grid
```

We get this output:
![image](https://user-images.githubusercontent.com/79875276/169362474-9771cb39-f60b-46a3-8ca5-d03c17d247b6.png)

We can also call the data getter:

```python
grid.data
```

![image](https://user-images.githubusercontent.com/79875276/169363472-35fa7177-f8cf-46f2-8c85-a79af6a120c2.png)

3. Now drop all the rows in the dataframe and set using the data setter.

```python
grid.data = grid.data.drop(0)
```

We see that the DataGrid UI object reflects those changes (we only have the columns):

![image](https://user-images.githubusercontent.com/79875276/169362691-8dfc317a-031d-467a-8930-248fbbe32bef.png)

4. Now call the data getter:

```python
grid.data
```

5. See error

![image](https://user-images.githubusercontent.com/79875276/169362851-4324213e-825d-4370-9ba0-03135414a673.png)

**Expected behavior**
I expect an empty dataframe to be returned like this when calling the data getter:

![image](https://user-images.githubusercontent.com/79875276/169363037-b10e20b1-41cd-479a-868b-1a5e2606f2d6.png)

**Environment (please complete the following information):**
 - Operating System and Version: Ubuntu 20.04 WSL
 - Browser [e.g. Firefox, Safari] (if applicable): Chrome (Using Jupyter Lab)

Any help with this is much appreciated. Cheers!

**Comment by ibdafna** [comment]
Hello :wave: @ollyhensby and welcome to the `ipydatagrid` community! Many thanks for reporting - I'll get this fixed!

**Comment by ibdafna** [comment]
Closed via #358 Awesome work @ollyhensby 🙏 Thank you!
